ui/src/components/Canvas.tsx

```diff
@@ -494,6 +491,27 @@ const CodeNode = memo<Props>(function ({
     nodesMap.delete(id);
   };
 
+  // update the level of a node as well as it's all descendants
+  function updateLevel(id: string, level: number) {
+    const node = nodesMap.get(id);
+    if (node) {
+      (node as any).level = level;
+      node.style!.backgroundColor = level2color[level];
+      nodesMap.set(id, node);
+      getPod(id)?.children.forEach(({ id }) => updateLevel(id, level - 1));
+    }
+  }
+
+  const removeParentByNodeById = (id) => {
+    const node = nodesMap.get(id)
+    if(node?.parentNode){
+      node.parentNode = undefined
+      node.data!.parent = 'ROOT'
+      nodesMap.set(id,node)
+      updateLevel(node.id, (node as any).level - 1);
```

Comment:
This line update the level

Reply:
Yeah, thanks! I posted that comment to just re-answer Dr. Bao’s original question.